projects/opendr_ws_2/src/opendr_perception/opendr_perception/wave_detection_node.py

```diff
@@ -0,0 +1,348 @@
+        """
+        Creates a ROS2 Node for rule-based wave detection via pose estimation.
+        :param input_rgb_image_topic: Topic from which we are reading the input image
+        :type input_rgb_image_topic: str
+        :param input_pose_topic: Topic from which we are reading the input pose list
+        :type input_pose_topic: str
+        :param output_rgb_image_topic: Topic to which we are publishing the annotated image (if None, no annotated
+        image is published)
+        :type output_rgb_image_topic: str
+        :param detections_topic: Topic to which we are publishing the annotations (if None, no wave detection message
+        is published)
+        :type detections_topic:  str
+        :param device: device on which we are running inference ('cpu' or 'cuda')
+        :type device: str
+        :param performance_topic: Topic to which we are publishing performance information (if None, no performance
+        message is published)
+        :type performance_topic:  str
+        :param num_refinement_stages: Specifies the number of pose estimation refinement stages are added on the
+        model's head, including the initial stage. Can be 0, 1 or 2, with more stages meaning slower and more accurate
+        inference
+        :type num_refinement_stages: int
+        :param use_stride: Whether to add a stride value in the model, which reduces accuracy but increases
+        inference speed
+        :type use_stride: bool
+        :param half_precision: Enables inference using half (fp16) precision instead of single (fp32) precision.
+        Valid only for GPU-based inference
+        :type half_precision: bool
+        """
+        super().__init__('opendr_wave_detection_node')
+
+        # If input image topic is set, it is used for visualization
+        if input_rgb_image_topic is not None:
+            self.image_subscriber = self.create_subscription(ROS_Image, input_rgb_image_topic, self.image_callback, 1)
+            self.image_publisher = self.create_publisher(ROS_Image, output_rgb_image_topic, 1)
+
+            # Initialize the pose estimation learner needed to run pose estimation on the image
+            self.pose_estimator = LightweightOpenPoseLearner(device=device, num_refinement_stages=num_refinement_stages,
+                                                             mobilenet_use_stride=use_stride,
+                                                             half_precision=half_precision)
+            self.pose_estimator.download(path=".", verbose=True)
+            self.pose_estimator.load("openpose_default")
+
+            if performance_topic is not None:
+                self.image_performance_publisher = self.create_publisher(Float32, performance_topic + "/image", queue_size=1)
+            else:
+                self.image_performance_publisher = None
+        else:
+            self.input_rgb_image_topic = None
+            self.image_publisher = None
```

Comment:
Do you also need to set the image subscriber to `None`?
```suggestion
            self.image_publisher = None
            self.image_subscriber = None
```